Source/inv.cpp

```diff
@@ -450,6 +450,99 @@ void DrawInvBelt()
 	}
 }
 
+BOOL IsEmpty(const ItemStruct &item)
+{
+	return item._itype == ITYPE_NONE;
+}
+
+BOOL CanEquip(const ItemStruct &item)
+{
+	return item._iLoc != ILOC_INVALID && item._iLoc != ILOC_NONE && item._iLoc != ILOC_UNEQUIPABLE && item._iStatFlag;
+}
+
+BOOL CanWield(int playerNumber, const ItemStruct &item)

```

Comment:
Could you add a breaf description of what this does. I'm unsure what the difference between this and CanEquip is just by the names.

Reply:
I used the term `wield` on purpose since it means "using with your hands" in English (basically). It is a more specific version of `equip` in Diablo's case.

So `CanEquip(item, location_left_hand)` means `left_hand.IsEmpty && CanWield(item)`

I can add comments on it, but I thought it wasn't necessary if you understand the term, felt self-explanatory to me. Let me know if you want me to add comments though.

Reply:
I get that, but I still find it a bit ambigious if I should not use weapons with `CanEquip`, or will work for both. Basically, it's not clear if `CanEquip` is just a poorly named `CanWear` since there is a special function for weapons. If `CanEquip` works for weapons, then why would I ever use `CanWield`?

Names of functions definatly should be descriptive, but if we at some point come in doubth about what the 50 lines of code do, it emensly helpful to have a short sentince that explains the context.

Reply:
I disagree on this being ambiguous but I understand this is highly subjective. Will add the comments as suggested.

Reply:
Added comments on all new methods.